src/index.js

```diff
@@ -1,19 +1,21 @@
 const mix = require('laravel-mix')
 const forIn = require('lodash/forIn')
 const jsonfile = require('jsonfile')
-const removeHashFromKeyRegex = /\.(.+)\.(.+)$/g
 const ExtractTextPlugin = require('extract-text-webpack-plugin')
 
 class VersionHash {
     register(options = {}) {
         this.options = Object.assign(
             {
-                length: 6
+                length: 6,
+                delimiter: '.'
             },
             options
         )
 
+        let delimiter = this.options.delimiter.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&')
```

Comment:
am not quit sure what is this for ?

Reply:
It's to escape character in the regular expression; in most cases people would probably use `.`, `-` or the the like. But if they would input `dada` for any reasy then the expression would catch the `\d` and try to match it with a digit, which would't match the correct pattern. 

Hope that makes sense =/ Sadly Javascript like this isn't my strongest suite, I'm sure there are more clever ways to handle those cases.

Stack Overflow discussion: https://stackoverflow.com/a/9310752/684195